### PR TITLE
fix: add the range search params check in proxy

### PR DESCRIPTION
**Pull request by smellthemoon**
if check in Segcore, will not do the it when not insert data.
so, check "radius" and "range_filter" in proxy.
related with #30365

### CODE REVIEW

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

**Review by congqixia — COMMENTED** [review]
why shall we remove Range search param check logic in segcore? we cannot make sure that segcore will not be used alone and secgore shall not reply on the caller always provided valid input

File: internal/proxy/task_search.go [hunk 1]
```diff
@@ -884,6 +890,57 @@ func reduceSearchResultData(ctx context.Context, subSearchResultData []*schemapb
...
+	// not pass range_filter, no need to check
+	if !ok {
+		return nil
+	}
+	err = json.Unmarshal(*data["range_filter"], &params.rangeFilter)
+	if err != nil {
+		return merr.WrapErrParameterInvalidMsg("must pass numpy type for range_filter")
+	}
+
+	if metricType == "IP" || metricType == "COSINE" {
```
Comment (congqixia) [inline internal/proxy/task_search.go:930]: use `metric.PositivelyRelated` function

File: internal/proxy/task_search.go [hunk 2]
```diff
@@ -884,6 +890,57 @@ func reduceSearchResultData(ctx context.Context, subSearchResultData []*schemapb
...
+		// no search params, no need to check
+		return nil
+	}
+	var data map[string]*json.RawMessage
+	err := json.Unmarshal([]byte(str), &data)
+	if err != nil {
+		log.Info("json Unmarshal fail when checkRangeSearchParams")
+		return err
+	}
+	_, ok := data["radius"]
```
Comment (congqixia) [inline internal/proxy/task_search.go:909]: add const for json key

File: internal/proxy/task_search.go [hunk 3]
```diff
@@ -884,6 +890,57 @@ func reduceSearchResultData(ctx context.Context, subSearchResultData []*schemapb
...
+	if err != nil {
+		return merr.WrapErrParameterInvalidMsg("must pass numpy type for radius")
+	}
+
+	_, ok = data["range_filter"]
+	// not pass range_filter, no need to check
+	if !ok {
+		return nil
+	}
+	err = json.Unmarshal(*data["range_filter"], &params.rangeFilter)
```
Comment (congqixia) [inline internal/proxy/task_search.go:925]: ditto, add or use const

**Comment by smellthemoon** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**